octodns/processor/filter.py

```diff
@@ -125,6 +127,35 @@ def _process(self, zone, *args, **kwargs):
     process_target_zone = _process
 
 
+class _NetworkValueBaseFilter(BaseProcessor):
+    def __init__(self, name, _list):
+        super().__init__(name)
+        self.networks = []
+        for value in _list:
+            try:
+                self.networks.append(ip_network(value))
+            except ValueError:
+                raise ValueError(f'{value} is not a valid CIDR to use')
+
+    def _process(self, zone, *args, **kwargs):
+        for record in zone.records:
+            if record._type not in ['A', 'AAAA']:
+                continue
```

Comment:
Kind of funky case here since these records technically don't match the filter, but if this called `doesnt_match` then they'd all get removed. I think this is the best path, to not even consider them, but it probably needs to be documented in the docstrings below that the filters only operate on `A` and `AAAA` records and that all other types will be left as-is.

IF someone actually wanted to get rid of the other types they could use the type based filters for that.

Reply:
I've added this comment